addons/mail/models/mail_message.py

```diff
@@ -911,9 +911,9 @@ def _message_fetch(self, domain, max_id=None, min_id=None, limit=30):
             domain = expression.AND([domain, [('id', '<', max_id)]])
         if min_id:
             domain = expression.AND([domain, [('id', '>', min_id)]])
-        return self.search(domain, limit=limit).message_format()
+        return self.search(domain, limit=limit).message_format(format_attachments_token='if-author' if self.env.user.share else False)
```

Comment:
Nitpicking: the new parameter has 2 possible values: `'if-author'` and `False`. Why not a simple boolean?

Reply:
Because `True` looks like it should always be set, but we only want to set it if the current user is author.
It could be `True` if the param was named `format_attachments_token_if_author` but that looked a bit too long.
I don't really mind either way if there is a strong consensus for one or the other.

Reply:
Given this is a Python -> Python communication, maybe an `Enum` would be a good option? Or a signaling `object()` checked by identity? That's a pretty common pattern.

Reply:
I've never used those before, so just `IF_AUTHOR = object()` at the top of the file to avoid using hardcoded string?